functions-and-scope.js

```diff
@@ -1,47 +1,86 @@
+
 // Je gaat functies schrijven die we kunnen hergebruiken om een lijst met eindcijfers van studenten te checken. Je zult over de cijfers heen moeten itereren (hoe pak je dat aan?),
 // maar ook een manier moeten vinden om hetgeen dat je verzamelt ergens te bundelen. Op deze manier zul je ontdekken hoe je omgaat met scope. Pak vooral het hoofdstuk op EdHub over for-loops er nog eens bij!
 // Tip: je mag hier geen ingebouwde object methoden gebruiken, dus daar hoef je niet naar te kijken.
 
-const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
-
 /* Opdracht  1: Cum Laude */
 
 /* 1a: Script schrijven  */
 // De administratie moet weten hoeveel studenten er dit blok cum laude zijn afgestudeerd (8 of hoger). Daar moeten namelijk speciale diploma's voor besteld worden.
 // Schrijf de stapjes om dit te kunnen checken eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array checken op deze conditie?
+//met een for loop
 // * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?
+//door over de lengte van de array te itereren
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
+//door een lege globale variable aan te maken om ze in op te slaan
 // Log het antwoord in de terminal.
-
 // ---- Verwachte uitkomst: 6
 
+const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
+
+let cumlaudeGrades = [];
+
+for (let i = 0; i < grades.length; i++) {
+    if (grades[i] >= 8) {
+        cumlaudeGrades.push(grades[i]);
+    }
+}
+
+let cumlaudeStudents = cumlaudeGrades.length;
+console.log(cumlaudeStudents);
 
 /*  1b: Omschrijven tot een herbruikbare functie   */
 // Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
+function cumLaude(array) {
+    let cumlaudeGrades = [];
+    for (let i = 0; i < array.length; i++) {
+        if (array[i] >= 8) {
+            cumlaudeGrades.push(array[i]);
+            if (cumlaudeGrades.length === 0) {
+                return 0;
+            }
+
+        }
+    }
+    let cumlaudeStudents = cumlaudeGrades.length;
```

Comment:
als je cumlaudeGrades.length terug geeft heb je hetzelfde. Dan hoef je zoals hierboven ook niet een 2de if te maken. Dan heb je een kortere code. 
VOORBEELD (mijnCode):

function cumLaude( cijfers ) {
    let cumLaudeNumbers = [];

    for( let i = 0; i < cijfers.length; i++ ) {
        if ( cijfers[i] >= 8 ) {
            cumLaudeNumbers.push( cijfers[i] );
        }
    }
    return cumLaudeNumbers.length;
}

console.log( cumLaude(grades) );
console.log( cumLaude([6, 4, 5]) );
console.log( cumLaude([8, 9, 4, 6, 10]) );